domain/src/main/java/org/oppia/domain/audio/AudioPlayerController.kt

```diff
@@ -0,0 +1,174 @@
+package org.oppia.domain.audio
+
+import android.content.Context
+import android.media.MediaPlayer
+import android.net.Uri
+import androidx.annotation.VisibleForTesting
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.MutableLiveData
+import kotlinx.coroutines.CoroutineDispatcher
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Job
+import kotlinx.coroutines.delay
+import kotlinx.coroutines.launch
+import org.oppia.util.data.AsyncResult
+import org.oppia.util.threading.BackgroundDispatcher
+import java.util.concurrent.TimeUnit
+import java.util.concurrent.locks.ReentrantLock
+import javax.inject.Inject
+import kotlin.concurrent.withLock
+
+/** Controller which provides audio playing capabilities.
+ * [initializeMediaPlayer] should be used to download a specific audio track.
+ * [releaseMediaPlayer] should be used to clean up the controller's resources.
+ * See documentation for both to understand how to use them correctly. */
+class AudioPlayerController @Inject constructor(
+  val context: Context, @BackgroundDispatcher private val backgroundDispatcher: CoroutineDispatcher
+) {
+
+  inner class AudioMutableLiveData : MutableLiveData<AsyncResult<PlayProgress>>() {
+    override fun onActive() {
+      super.onActive()
+      observerActive = true
+    }
+
+    override fun onInactive() {
+      super.onInactive()
+      observerActive = false
+      stopUpdatingSeekBar()
+    }
+  }
+
+  enum class PlayStatus { PREPARED, PLAYING, PAUSED, COMPLETED}
```

Comment:
Nit: space before last ``}``.

Please add documentation to PlayStatus and PlayProgress explaining what they represent and how they can be used.

Also, do we still need a ``PREPARED`` state or could we maybe default to ``PLAYING``?

Reply:
I think it would be better to have a PREPARED state because it's the first time the fragment will know about the audio's duration and can set the seek bar max value accordingly.